src/battle_controller_player.c

```diff
@@ -1435,19 +1435,22 @@ static void Task_GiveExpToMon(u8 taskId)
         u8 level = GetMonData(mon, MON_DATA_LEVEL);
         u32 currExp = GetMonData(mon, MON_DATA_EXP);
         u32 nextLvlExp = gExperienceTables[gSpeciesInfo[species].growthRate][level + 1];
+        u32 expAfterGain = currExp + gainedExp;
         u32 oldMaxHP = GetMonData(mon, MON_DATA_MAX_HP);
 
-        if (currExp + gainedExp >= nextLvlExp)
+        if (expAfterGain >= nextLvlExp)
         {
-            SetMonData(mon, MON_DATA_EXP, &nextLvlExp);
+            bool32 combineLevelUpMessages = B_LEVEL_UP_NOTIFICATION >= GEN_9;
```

Comment:
We have basically no precedent for storing config comparisons, so I would replace this (and all other instances) with just checking the actual config when relevant as the compiler will take care of the rest

Reply:
[`32b1492` (#4901)](https://github.com/rh-hideout/pokeemerald-expansion/pull/4901/commits/32b149202a71ab7e5e4acd5aceae0eba6763415d)